spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/store/MongoDbMessageStore.java

```diff
@@ -511,18 +512,21 @@ private void updateGroup(Object groupId, Update update) {
 	private long getNextId() {
 		Query query = Query.query(Criteria.where("_id").is(SEQUENCE_NAME));
 		query.fields().include(SEQUENCE);
-		return ((Number) this.template.findAndModify(query,
-						new Update().inc(SEQUENCE, 1L),
-						FindAndModifyOptions.options().returnNew(true).upsert(true),
-						Map.class, this.collectionName)
-				.get(SEQUENCE))  // NOSONAR - never returns null
+		Map<?, ?> result = this.template.findAndModify(query,
+				new Update().inc(SEQUENCE, 1L),
+				FindAndModifyOptions.options().returnNew(true).upsert(true),
+				Map.class, this.collectionName);
+		Objects.requireNonNull(result);
+		return ((Number) Objects.requireNonNull(result
+				.get(SEQUENCE)))
 				.longValue();
 	}
 
 	@SuppressWarnings(UNCHECKED)
 	private static void enhanceHeaders(MessageHeaders messageHeaders, Map<String, Object> headers) {
 		Map<String, Object> innerMap =
 				(Map<String, Object>) new DirectFieldAccessor(messageHeaders).getPropertyValue(HEADERS);
+		Objects.requireNonNull(innerMap);
```

Comment:
Same here.
Probably `NullAway` suppression is the way to go for these methods .